fs/backgroundfetcher/resolver.go

```diff
@@ -65,6 +67,8 @@ func (b *base) Closed() bool {
 type sequentialLayerResolver struct {
 	*base
 	nextSpanFetchID compression.SpanID
+	// timestamp when background fetch for the layer starts
+	start time.Time
```

Comment:
This should be part of `base` since we want it for each `Resolver`, not just `sequentialLayerResolver`. 

How about adding something like these methods to the interface

```
type Resolver interface {

      Start() // Starts the timer
      
      Duration() time.Time // Get the time
     
}
```

And do `resolver.Start()` in `backgroundfetcher.Add()`.

Reply:
Why do we need ``Resolver`` interface? ``sequentialLayerResolver`` is the only implementation of ``Resolver`` .

Reply:
For now. In the future, we may have different implementation for Resolver. (For example, a resolver with an ordered list of spans to fetch).

Reply:
Decided to move ``start`` to ``base`` but keep where ``start`` timestamp is recorded unchanged.  The timestamp when span 0 starts being fetched is a more accurate measurement of when resolving for the layer starts than when the resolver is added to work queue.